inst/include/interface/rcpp/rcpp_objects/rcpp_recruitment.hpp

```diff
@@ -15,219 +15,317 @@
-    if (this->log_sigma_recruit.estimated) {
-      if (this->log_sigma_recruit.is_random_effect) {
-        d2->RegisterRandomEffect(b2->log_sigma_recruit);
-      } else {
-        d2->RegisterParameter(b2->log_sigma_recruit);
-      }
-    }
-    // add to Information
-    d2->recruitment_models[b2->id] = b2;
-
-    // third-order derivative
-    std::shared_ptr<fims::Information<TMB_FIMS_THIRD_ORDER> > d3 =
-        fims::Information<TMB_FIMS_THIRD_ORDER>::GetInstance();
-
-    std::shared_ptr<fims::SRBevertonHolt<TMB_FIMS_THIRD_ORDER> > b3 =
-        std::make_shared<fims::SRBevertonHolt<TMB_FIMS_THIRD_ORDER> >();
-
-    // set relative info
-    b3->id = this->id;
-    b3->steep = this->steep.value;
-    if (this->steep.estimated) {
-      if (this->steep.is_random_effect) {
-        d3->RegisterRandomEffect(b3->steep);
-      } else {
-        d3->RegisterParameter(b3->steep);
-      }
-    }
-    b3->rzero = this->rzero.value;
-    if (this->rzero.estimated) {
-      if (this->rzero.is_random_effect) {
-        d3->RegisterRandomEffect(b3->rzero);
-      } else {
-        d3->RegisterParameter(b3->rzero);
-      }
-    }
-    b3->log_sigma_recruit = this->log_sigma_recruit.value;
-    if (this->log_sigma_recruit.estimated) {
-      if (this->log_sigma_recruit.is_random_effect) {
-        d3->RegisterRandomEffect(b3->log_sigma_recruit);
-      } else {
-        d3->RegisterParameter(b3->log_sigma_recruit);
-      }
+
+     virtual double evaluate_nll() {
+    fims::SRBevertonHolt<double> NLL;
+
+    NLL.log_sigma_recruit = this->log_sigma_recruit.value;
+    NLL.recruit_deviations.resize(
+        deviations.size());  // Vector from TMB
+    NLL.recruit_bias_adjustment.resize(
+        recruit_bias_adjustment.size());  // Vector from TMB
+    for (int i = 0; i < deviations.size(); i++) {
+      NLL.recruit_deviations[i] = deviations[i];
+      NLL.recruit_bias_adjustment[i] = recruit_bias_adjustment[i];
     }
-    // add to Information
-    d3->recruitment_models[b3->id] = b3;
+    Rcout << "Rec devs being passed to C++ are " << deviations
+          << std::endl;
+
+    Rcout << "Rec bias adj being passed to C++ are " << recruit_bias_adjustment
+          << std::endl;
 
-    return true;
+    NLL.use_recruit_bias_adjustment = this->use_bias_correction;
+    NLL.estimate_recruit_deviations = this->estimate_deviations;
+    // NLL.PrepareConstrainedDeviations();
+    // NLL.PrepareBiasAdjustment();
+    return NLL.evaluate_nll();
   }
+
+    /** @brief this adds the parameter values and derivatives to the TMB model
+     * object */
+    virtual bool add_to_fims_tmb() {
+        // base model
+        std::shared_ptr<fims::Information<TMB_FIMS_REAL_TYPE> > d0 =
+                fims::Information<TMB_FIMS_REAL_TYPE>::GetInstance();
+
+        std::shared_ptr<fims::SRBevertonHolt<TMB_FIMS_REAL_TYPE> > b0 =
+                std::make_shared<fims::SRBevertonHolt<TMB_FIMS_REAL_TYPE> >();
+
+        // set relative info
```

Comment:
suggestion: make this comment more informative regarding the different types of derivatives.